src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js

```diff
@@ -62,27 +64,58 @@ const ActiveState = ({
                         </li>
                     );
                 case 'task':
-                    return item.task_assignment_collection.total_count ? (
-                        <li key={type + id} className="bcs-activity-feed-task">
-                            <Task
-                                {...item}
-                                currentUser={currentUser}
-                                getAvatarUrl={getAvatarUrl}
-                                getMentionWithQuery={getMentionWithQuery}
-                                getUserProfileUrl={getUserProfileUrl}
-                                mentionSelectorContacts={mentionSelectorContacts}
-                                onAssignmentUpdate={onTaskAssignmentUpdate}
-                                onDelete={onTaskDelete}
-                                onEdit={onTaskEdit}
-                                permissions={{
-                                    can_delete: true,
-                                    can_edit: true,
-                                }}
-                                // permissions are not part of task API so hard code to true
-                                translations={translations}
+                    return (
+                        <div>
+                            <FeatureFlag
+                                feature="activityFeed.tasks.newCards"
+                                disabled={() =>
+                                    item.task_assignment_collection &&
+                                    item.task_assignment_collection.total_count && (
+                                        <li key={type + id} className="bcs-activity-feed-task">
+                                            <Task
```

Comment:
The `<Task>` and `<TaskNew>` markup looks very similar between enabled/disabled. Is it possible for us to limit the feature flag to the class name to avoid the duplication?

Reply:
I'm struggling to see a clean way to pull out the task_assignment_collection and the new task layout logic into the same computation using isFeatureEnabled. The assignment collection check has to be contained within the `task` case as well as the feature flag.

Reply:
The new changes look good to me. We can just nuke the old `<Task>` component once the new stuff is out 100%.